subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/result/ComponentResultSerializer.java

```diff
@@ -60,6 +61,7 @@ public void write(Encoder encoder, ComponentResult value) throws IOException {
         componentIdSerializer.write(encoder, value.getComponentId());
         encoder.writeString(value.getVariantName().getDisplayName());
         attributeContainerSerializer.write(encoder, value.getVariantAttributes());
+        encoder.writeNullableString(value.getRepositoryId());
```

Comment:
@bigdaz @melix doesn't this get serialized and deserialized across build invocations? If so, we might have a problem here that this ID may no longer match after serialization/deserialization. 

For example, if the order of the repos changes between invocations the IDs could be different, IIUC.

Reply:
The repository ID is independent of order, and the serialization is only used within one build (it's used to serialize the graph, so that we can read it back if we need to re-resolve later), so I think we're safe.